xonsh/environ.py

```diff
@@ -120,6 +121,7 @@ def to_debug(x):
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
     'XONSH_HISTORY_SIZE': (is_history_tuple, to_history_tuple, history_tuple_to_str),
     'XONSH_LOGIN': (is_bool, to_bool, bool_to_str),
+    'XONSH_SHOW_TRACEBACK': (is_tb_opt, to_tb_tuple, tb_tuple_to_str),
```

Comment:
If I understand correctly, if those checks are put directly inside `print_exception` and `XONSH_SHOW_TRACEBACK` is removed from `DEFAULT_ENSURES`, a user won't know that `XONSH_SHOW_TRACEBACK` is invalid until an exception is raised (please excuse any possible design misunderstandings from my side, as I've only recently gotten acquainted with the codebase).

On the other hand, If you were making a point strictly about the fact that the representation is changed, I too think we can manage with storing just the `repr` of the option.